tests/test_reinforce_algorithm.py

```diff
@@ -0,0 +1,168 @@
+import itertools
+
+import numpy as np
+import pytest
+
+from reinforcement.algorithm.reinforce import Reinforce
+from reinforcement.trajectories import TrajectoryBuilder, Trajectories
+
+CALL_ORDER = list()
+
+
+class OperationsSim:
+    @staticmethod
+    def reduce_mean(array):
+        return np.mean(array)
+
+    @staticmethod
+    def log(array):
+        return np.log(array).T
+
+
+class PolicySpy:
+    def __init__(self):
+        self.received_returns = None
+        self.received_loss_signal = None
+        self._calc = None
+
+    def set_signal_calc(self, calc):
+        self._calc = calc
+
+    def fit(self, batch):
+        CALL_ORDER.append(self.fit)
+        self.received_returns = list(batch.returns)
+        sim = OperationsSim()
+        losses = [self._calc(sim, a, r) for a, r in zip(batch.actions, batch.returns)]
```

Comment:
Same as for the reward signal. The next merge should contain tests that hopefully can detect this issue reliably.